packages/main/src/pages/preferences/index.js

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useEffect } from "react";
+import { BsBell, BsGlobe, BsLayoutTextSidebarReverse } from "react-icons/bs";
+import { IoColorFilterOutline } from "react-icons/io5";
+import { FiMonitor } from "react-icons/fi";
+import { MdPrivacyTip } from "react-icons/md";
+
+// components
+import Header from "./components/Header";
+import Accessibility from "./components/Accessibility";
+import Appearance from "./components/Appearance";
+import LangRegion from "./components/LangRegion";
+import Notifications from "./components/Notifications";
+import Privacy from "./components/Privacy";
+import Sidebar from "./components/Sidebar";
+
+import styles from "./assets/css/main.css";
+
+export default function index() {
+  const [showAccessibility, setShowAccessibility] = useState(false);
+  const [showAppearance, setShowAppearance] = useState(false);
+  const [showLangRegion, setShowLangRegion] = useState(false);
+  const [showNotifications, setShowNotifications] = useState(true);
+  const [showPrivacy, setShowPrivacy] = useState(false);
+  const [showSidebar, setShowSidebar] = useState(false);
+  const [preference, setPreference] = useState("notifications");
+
+  const accessibility = () => {
```

Comment:
what happens when someone wants to add a new preference type?
Do they have to add a setState to all the functions?

change this and use a single object to store all the states
you can now loop through the object and change everything to false except the one you want to update